File: LeetCode/167 - Two Sum II - Input array is sorted/solve4.cc
class Solution {
public:
    
    vector<int> twoSum(vector<int>& numbers, int target) {
        int lo = 0, hi = numbers.size()-1;
        while (lo < hi) {
            int sum = numbers[lo] + numbers[hi];
            
            if (sum > target) --hi;
            else if (sum < target) ++lo;
            else {
                std::vector<int> ans = {lo+1, hi+1};
                return ans;
            }
        }
        return std::vector<int>();
    }
};

File: LeetCode/1. Two Sum/solve4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, n in enumerate(nums):
            other = target-n
            if other in d:
                return [d[other], i]
            d[n] = i
        return []
        
